app/models/candidate_interface/training_with_a_disability_form.rb

```diff
@@ -0,0 +1,32 @@
+module CandidateInterface
+  class TrainingWithADisabilityForm
+    include ActiveModel::Model
+
+    attr_accessor :disclose_disability, :disability_disclosure
+
+    validates_inclusion_of :disclose_disability, in: [true, false, 'true', 'false']
+
+    def self.build_from_application(application_form)
+      new(
+        disclose_disability: application_form.disclose_disability,
+        disability_disclosure: application_form.disability_disclosure,
+      )
+    end
+
+    def save(application_form)
+      return false unless valid?
+
+      # explicitly null-out the text field if the user said 'No'
```

Comment:
Perhaps we can avoid some of the confusion here by using different values in the form. 

If we make the form values `yes` and `no`, we explicitly translate them to true/false. I think this is how it works in the personal details form:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/a22b836482ad2ba579f0711c645d796c815450db/app/models/candidate_interface/personal_details_form.rb#L33

Reply:
we still have to explicitly handle `nil`, and there's an interdependence between the two values - but I've refactored as you suggested, and it does make it clearer 👍